CONTRIBUTING.md

```diff
@@ -48,4 +49,30 @@ In order for code to be merged into master it must
 The reviewers are expected to look out for the items above before approving a merge
 request.
 
-
+#### Formatting Code
+We use clang-format to automatically format the code. If you have clang-format installed
+locally, you can always execute `make format` or `cmake --build . --target format` from
+your build directory to automatically format the code.
+
+If you don't have clang-format installed locally, our "Hermes" automation can always
+format the code for you. Just create the following comment in your PR, and Hermes will
+format the code and automatically commit it:
+```
+@par-hermes format
+```
+
+After Hermes formats your code, remember to run `git pull` to update your local tracking
+branch.
+
+If you'd like Hermes to amend the formatting to the latest commit, you can use the
+`--amend` option, like so:
+```
+@par-hermes format --amend
```

Comment:
Should we potentially add a big warning that one should only ever do this right after a push (I'm thinking about the situation where more people work on the same branch and things may start to diverge if amends, fetches, pulls,reset are not coordinated).

Reply:
I'll add a warning - yeah, if you're not aware of the implications of amending, it could be misused.